tools/build-rpms.py

```diff
@@ -0,0 +1,254 @@
+        f.write("[rpmbuilder]\n")
+        for i in instances:
+            arch = arch_rename_aws.get(i.architecture, i.architecture)
+            f.write(
+                f"{i.public_ip_address} " +
+                ' '.join(
+                    [
+                        f"ansible_ssh_private_key_file={keypath}",
+                        f"ARCH={arch}"]) +
+                "\n")
+
+
+def run_ansible():
+    osbuild_commit = json.load(
+        open(f"{base_dir}/Schutzfile"))["rhel-8.6"]["dependencies"]["osbuild"]["commit"]
+
+    return run_command(["ansible-playbook",
+                        # "-vvv",
+                        "--ssh-extra-args", "-o ControlPersist=no -o StrictHostKeyChecking=no -o ServerAliveInterval=5",
+                        "-i", f"{base_dir}/tools/appsre-ansible/inventory",
+                        f"{base_dir}/tools/appsre-ansible/rpmbuild.yml",
+                        "-e", f"COMPOSER_COMMIT={commit}",
+                        "-e", f"OSBUILD_COMMIT={osbuild_commit}",
+                        "-e", f"RH_ACTIVATION_KEY={os.environ['RH_ACTIVATION_KEY']}",
+                        "-e", f"RH_ORG_ID={os.environ['RH_ORG_ID']}"])
+
+
+def stage_generate_rpms(arch):
+    keyname, key_material = stage("Create keypair", (), create_keypair)
+    instances = stage("Create EC2 instances", (),
+                      create_ec2_instances, arch, keyname)
+    print(instances)
+    stage("Setup ansible", (), setup_ansible, key_material, instances)
+    stage("Run Ansible playbook", (), run_ansible)
+
+
+def check_env():
+    required_envvars = ["RH_ORG_ID", "RH_ACTIVATION_KEY"]
+    if not all((i in os.environ for i in required_envvars)):
+        msg_error(
+            f"At least one of the required environment variables is missing: {required_envvars}")
+        sys.exit(1)
+
+
+def check_params():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "--dry-run",
+        help="Only print steps without doing actions",
+        action='store_true')
+
+    parser.add_argument('--base-dir', type=pathlib.Path, required=True,
+                        help='Base directory for creating files')
+
+    parser.add_argument('--commit', type=str, required=True,
+                        help='Commit SHA')
+
+    parser.add_argument('arch', type=str, nargs='+', choices=arch_info.keys(),
+                        help='Architectures to build images for')
+
+    args = parser.parse_args()
+
+    global is_dryrun, base_dir, commit
+
+    def is_dryrun(): return args.dry_run
+    base_dir = args.base_dir
+    commit = args.commit
+
+    return args
+
+
+def print_params(p):
+    if is_dryrun():
+        msg_info("Running in dryrun mode")
+
+    msg_info(f"Building {p.arch} RPMs with base-dir {p.base_dir}")
+
+
+if __name__ == "__main__":
+    exitcode=0
+
+    check_env()
+    args = check_params()
+    print_params(args)
+
+    try:
+        stage_generate_rpms(args.arch)
+    except Exception as e:
+        msg_error(e)
```

Comment:
I think that you could not catch the exception here at all and do a `try: ... finally: ...` thing. The except part here only prints the traceback and sets the exit code, this is already done automatically when an exception is raised.